post-processing/streamlit_post_processing.py

```diff
@@ -169,6 +177,23 @@ def axis_select(label: str, axis: dict):
 
     # units select
     units_select(label, axis)
+
+    # FIXME: add ability to use a custom value for only one of min or max
+    disable_user_range = plot_type != 'line'
+    use_default_ranges = st.checkbox("default axis range",
+                                     axis.get("range").get("use_default") if not disable_user_range else True,
+                                     key="{0}_axis_range_use_default".format(label),
+                                     disabled=disable_user_range,
+                                     help="{0} {1}".format(
+                                         "Uncheck to assign custom minimum and maximum values to axis.",
+                                         "Custom ranges are not implemented for datetime types."))
+    if not use_default_ranges:
+        axis_range_min, axis_range_max = st.columns(2)
+        with axis_range_min:
+            st.number_input("{0}-axis minimum".format(label), key="{0}_axis_range_min".format(label))
+        with axis_range_max:
+            st.number_input("{0}-axis maximum".format(label), key="{0}_axis_range_max".format(label))
```

Comment:
```suggestion
        with axis_range_min:
            st.number_input("{0}-axis minimum".format(label),
                            value=(axis["range"].get("min") if axis.get("range").get("min")
                                   else get_axis_min_max(df[mask], axis)[0]),
                            key="{0}_axis_range_min".format(label))
        with axis_range_max:
            st.number_input("{0}-axis maximum".format(label),
                            value=(axis["range"].get("max") if axis.get("range").get("max")
                                   else get_axis_min_max(df[mask], axis)[1]),
                            key="{0}_axis_range_max".format(label))
```
Yesterday I thought it would be nice if the initial values of these number inputs were to correspond to the default axis values generated, but I didn't quite have enough time to craft a proper addition. I think something like what I've put in the suggestion above would work, but I've only had the chance to test it haphazardly and I know that using `axis.get(...).get(...)` can result in unwanted behaviour.

I'd also recommend adding in a line for `mask = st.session_state.post.mask` somewhere at the top of this function, preferably under `df = st.session_state.post.df`, as well as a `from plot_handler import get_axis_min_max` above the `PostProcessing` import.

If this turns out to be too much of a pain to get working it can become a FIXME instead.